src/ConductorSharp.Engine/Service/DeploymentService.cs

```diff
@@ -50,7 +51,10 @@ public async Task Remove(Deployment deployment)
                 var oldDefinition = await _metadataService.GetWorkflowAsync(definition.Name, definition.Version);
 
                 if (oldDefinition?.Name != null)
-                    await _metadataService.DeleteWorkflowAsync(definition.Name, definition.Version);
+                    await _metadataService.DeleteWorkflowAsync(
+                        definition.Name,
+                        definition.Version ?? throw new InvalidOperationException($"Workflow {definition.Name} cannot be null")
```

Comment:
` throw new InvalidOperationException($"Workflow {definition.Name} version cannot be null")`